GTScript-Parallel-model.md

```diff
@@ -103,3 +103,175 @@ for k in range(start, end):
         u[i, j, k] = 1
     parfor [i_start:i_end, j_start:j_end]:
         b[i, j, k] = u[i-2,j,k] + u[i+1,j,k] + u[i,j-1,k] + u[i,j-2,k]
+```
+
+## Conditionals
+
+GTScript supports 2 kinds of conditionals:
+
+- conditionals on scalar expressions or compile-time constants 
+- conditionals on field expressions (meaning a condition on the field value at the gridpoint of the current iteration or at a local offset of it)
+
+### Conditionals on scalar expressions or compile-time constants
+
+- Conditions in `if` statements are evaluated at run-time and are required to be of scalar type, e.g. scalar variables (not fields) or scalar expressions (no fields involved).
+- Each statement inside the if and else branches is executed as a `parfor` loop (same as statements outside of branches).
+- There is no restriction on the body of the statement.
+- For consistency sake with the following we use the condition inside the loop as a concept (the implementation detail to not do this should not matter here)
+
+Rationale:
+- Adding an `if True:` does not change the behavior at all.
+- The way the programmer thinks about execution is the same if within an if or not.
+- This makes function calls and scoped code behave the same.
```

Comment:
There is an implication for runtime if's, since the extents of the statements in the if and the else can potentially be different. But this is an implementation detail, I guess.